### PR TITLE
ROC-5993 Generate and download the Review Order PDF

### PR DESCRIPTION
**Pull request by simad**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-5993

### Change description ###

Adding a service to generate the review order pdf. This uses the new generic endpoint.

### Dependencies ###
This needs CCD definitions in this PR released first.

https://github.com/hmcts/cmc-ccd-domain/pull/65
 
**Does this PR introduce a breaking change?** (check one with "x")

```
[] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ReviewOrderContentProvider.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    public Map<String, Object> createContent(Claim claim) {
+        ReviewOrder reviewOrder = claim.getReviewOrder().orElseThrow(IllegalStateException::new);
+        PersonContent party = reviewOrder.getRequestedBy().equals(ReviewOrder.RequestedBy.DEFENDANT) ?
+           PersonContent.builder()
+               .address(claim.getClaimData().getDefendant().getAddress())
+                .fullName(claim.getClaimData().getDefendant().getName()).build() :
+            PersonContent.builder()
+            .address(claim.getClaimData().getClaimant().getAddress())
+            .fullName(claim.getClaimData().getClaimant().getName()).build();
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ReviewOrderContentProvider.java]: Looks like the code formatter hasn't kicked in - please run manually

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/models/PersonContent.java [hunk 2]
```diff
@@ -1,9 +1,12 @@
 package uk.gov.hmcts.cmc.claimstore.services.staff.models;
 
+import lombok.Builder;
+import lombok.Getter;
 import uk.gov.hmcts.cmc.domain.models.Address;
 
+@Getter
+@Builder
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/models/PersonContent.java]: Much better!

File: src/main/resources/citizen/templates/document/reviewOrder.html [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    </tbody>
+  </table>
+</header>
+
+<div class="section">
+  <table>
+    <tbody>
+    <tr>
+      <td>
+        Party's name:
```
Comment (andrewwa-kainos) [inline src/main/resources/citizen/templates/document/reviewOrder.html]: Might want this changed to the pretty apostrophe `’`

File: src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ReviewOrderContentProvider.java (hunk 1 above)
Comment (simad) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ReviewOrderContentProvider.java]: 👍

File: src/main/resources/citizen/templates/document/reviewOrder.html (hunk 3 above)
Comment (simad) [reply src/main/resources/citizen/templates/document/reviewOrder.html]: 👍

File: src/main/resources/citizen/templates/document/reviewOrder.html [hunk 4]
```diff
@@ -88,7 +113,6 @@
   <table class="crestLogo">
     <tbody>
     <tr>
-      <td></td>
       <td class="claim-details">
         <h1>Request for review of order</h1>
         <p>In the County Court Business Centre Online Civil Money Claims</p>
```
Comment (sabahirfan) [inline src/main/resources/citizen/templates/document/reviewOrder.html]: Do we want to break this in two lines as we are doing with other similar ones?
```
<p>In the County Court Business Centre<br/>
 Online Civil Money Claims
</p>
```

File: src/main/java/uk/gov/hmcts/cmc/claimstore/events/revieworder/ReviewOrderEvent.java [hunk 5]
```diff
@@ -11,10 +11,16 @@
...
+    private final String authorisation;
     private final Claim claim;
 
-    public ReviewOrderEvent(Claim claim) {
+    public ReviewOrderEvent(String authorisation, Claim claim) {
         this.claim = claim;
+        this.authorisation = authorisation;
+    }
+
+    public String getAuthorisation() {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/events/revieworder/ReviewOrderEvent.java]: This is not needed as we are using `@Getter` annotation

**Pull request merged**